rogue-thi-app/pages/food.js

```diff
@@ -44,15 +46,26 @@ export default function Mensa () {
     allergenSelection,
     setShowFoodFilterModal
   } = useContext(FoodFilterContext)
-  const [foodDays, setFoodDays] = useState(null)
+  const [currentFoodDays, setCurrentFoodDays] = useState(null)
+  const [futureFoodDays, setFutureFoodDays] = useState(null)
   const [showMealDetails, setShowMealDetails] = useState(null)
+  const [page, setPage] = useState(0)
   const userKind = useUserKind()
 
   useEffect(() => {
     async function load () {
       try {
         const days = await loadFoodEntries(selectedRestaurants)
-        setFoodDays(days.slice(0, 5))
+
+        /**
+         * new Date(days[0].timestamp).getDay()
+         * api returns full next 2 weeks on weekends, so just `new Date()`
+         * to calculate the days till friday would be wrong (higher than 5)
+         * */
+        const daysTillFriday = 5 - new Date(days[0].timestamp).getDay() + 1
```

Comment:
Undefined error if no data is available, e.g. during the semester break or or if no restaurants are selected. Adding the `optional chaining` operator might help.